WordPress/WordPressTest/AccountSettingsServiceTests.swift

```diff
@@ -35,41 +20,44 @@ class AccountSettingsServiceTests: CoreDataTestCase {
         )
     }
 
-    private func managedAccountSettings() -> ManagedAccountSettings? {
-        contextManager.performQuery { context in
-            let request = NSFetchRequest<NSFetchRequestResult>(entityName: ManagedAccountSettings.entityName())
-            request.predicate = NSPredicate(format: "account.userID = %d", self.service.userID)
-            request.fetchLimit = 1
-            guard let results = (try? context.fetch(request)) as? [ManagedAccountSettings] else {
-                return nil
-            }
-            return results.first
-        }
-    }
-
     func testUpdateSuccess() throws {
-        stub(condition: isPath("/rest/v1.1/me/settings")) { _ in
-            HTTPStubsResponse(jsonObject: [String: Any](), statusCode: 200, headers: nil)
-        }
+        // We've seen some flakiness in CI on this test, and therefore are using a stub object rather than stubbing the HTTP requests.
+        // Since this approach bypasses the entire network stack, the hope is that it'll result in a more robust test.
+        //
+        // This is the second test in this class edited this way.
+        // If we'll need to update a third, we shall also take the time to update the rest of the tests.
+        let service = AccountSettingsService(
+            userID: 1,
+            remote: AccountSettingsRemoteInterfaceStub(updateSettingResult: .success(())),
+            coreDataStack: contextManager
+        )
+
         waitUntil { done in
-            self.service.saveChange(.firstName("Updated"), finished: { success in
+            service.saveChange(.firstName("Updated"), finished: { success in
                 expect(success).to(beTrue())
                 done()
             })
         }
+
         expect(self.managedAccountSettings()?.firstName).to(equal("Updated"))
     }
 
     func testUpdateFailure() throws {
-        stub(condition: isPath("/rest/v1.1/me/settings")) { _ in
-            HTTPStubsResponse(jsonObject: [String: Any](), statusCode: 500, headers: nil)
-        }
```

Comment:
Do you know what's causing the flakiness? I noticed this test case doesn't call `OHHTTPStubs.removeAllStubs` upon `tearDown`, which is likely a mistake and also likely to cause issues.

Reply:
I haven't looked into it. No.

This was my first attempt and it worked, so I was happy with it 😅. Worth digging in, though.